integration-test/common/src/test/kotlin/org/spekframework/spek2/integration/SetFeature.kt

```diff
@@ -1,20 +1,20 @@
 package org.spekframework.spek2.integration
 
 import org.spekframework.spek2.Spek
-import org.spekframework.spek2.style.gherkin.Feature
+import org.spekframework.spek2.style.gherkin.Given
 import kotlin.test.assertEquals
 import kotlin.test.assertFailsWith
 
 object SetFeature: Spek({
-    Feature("Set") {
+    Given("Set") {
         val set by memoized { mutableSetOf<String>() }
 
-        Scenario("is empty") {
+        When("is empty") {
             Then("should have a size of 0") {
                 assertEquals(0, set.size)
             }
 
-            And("should throw when first is invoked") {
+            Then("should throw when first is invoked") {
```

Comment:
Wait, so by default `memoized` is `TEST` scoped, doesn't this have to be nested now as @ming13 suggested?

Reply:
`Then` is a test. Not sure if I get what you're saying.

Reply:
(Not sure I'm completely following these changes, but)

>if you try to add stateful test, you'll see that it doesn't play well with Gherkin because memoized resets state for every Then

According to Gherkin, state should be shared across `Then`s, for example:

>There are examples when it not possible to split a scenario. Withdrawing money from a bank account is such an example.
The customer should get their money from an ATM machine, say £50
The customer account should be debited £50

(from http://www.thinkcode.se/blog/2016/06/22/cucumber-antipatterns)

So, for one `When` there are two `Then`s performing checks on a shared state.
And I'd say it's a pretty common situation.